### PR TITLE
Custom datasource support

### PR DESCRIPTION
**Pull request by webong**
This pull request makes the following changes:
- adds an `extend` method to DataSource Manager class
- adds a `customSources` property to DataSource Manager class
- updates `getSources` and `getEnabledSources` methods on DataSource Manager class
- updates `resolve` method on DataSource Manager class
- adds a new test case for custom sources to DataSourceManagerTest class

Test checklist:
- [ ]

- [ ] I certify that I ran my checklist

Fixes ushahidi/platform# .

Ping @ushahidi/platform

### CODE REVIEW

**Comment by webong** [comment]
@tuxpiper can you help review this

File: .gitignore [hunk 1]
```diff
@@ -52,3 +52,6 @@ modules/orm
 deployment.env
 docker.env
 vault.txt
+
+#Packages development
+packages
```
Comment (tuxpiper) [inline .gitignore:57]: What is the purpose of this `packages` folder?

File: src/App/DataSource/DataSourceManager.php [hunk 2]
```diff
@@ -38,6 +39,13 @@ class DataSourceManager
         'twitter' => Twitter\Twitter::class,
     ];
 
+    /**
+     * The registered custom data sources.
+     *
+     * @var array
+     */
+    protected $customSources = [];
```
Comment (tuxpiper) [inline src/App/DataSource/DataSourceManager.php:47]: Just asking about naming here... with "custom" do we mean datasources that are installed in as additional packages?

Comment (webong) [reply src/App/DataSource/DataSourceManager.php:47]: Yes ... data sources installed as a package

File: .gitignore (hunk 1 above)
Comment (webong) [reply .gitignore:57]: ohh ... I could remove this, I initially had a packages folder for package development in the platform root directory.

File: src/App/DataSource/DataSourceManager.php (hunk 2 above)
Comment (tuxpiper) [reply src/App/DataSource/DataSourceManager.php:47]: I would be inclined to call them "add-on" or "plug-in" datasources. To signify they are off the repo tree. Anyway, let's chat later, I'm really curious about the dynamic nature of the code below.

**Comment by webong** [comment]
@tuxpiper I have resolved all failing tests

**Pull request merged**